packages/aead/README.md

```diff
@@ -1,3 +1,257 @@
+
+## Usage
+
+### Implementing the Cipher Trait
+
+To use this crate, you need to implement the [`Cipher`] trait for your AEAD algorithm:
+
+```rust
+use vitaminc_aead::{Cipher, Unspecified, IntoAad, LocalCipherText};
+
+struct MyCipher {
+    // Your cipher implementation
+}
+
+impl Cipher for MyCipher {
+    fn encrypt_slice<'a, A>(
+        &self,
+        plaintext: &'a [u8],
+        aad: A,
+    ) -> Result<LocalCipherText, Unspecified>
+    where
+        A: IntoAad<'a>
+    {
+        // Your encryption implementation
+        todo!()
+    }
+
+    fn encrypt_vec<'a, A>(
+        &self,
+        plaintext: Vec<u8>,
+        aad: A,
+    ) -> Result<LocalCipherText, Unspecified>
+    where
+        A: IntoAad<'a>
+    {
+        // Your encryption implementation
+        todo!()
+    }
+
+    fn decrypt_vec<'a, A>(
+        &self,
+        ciphertext: LocalCipherText,
+        aad: A,
+    ) -> Result<Vec<u8>, Unspecified>
+    where
+        A: IntoAad<'a>
+    {
+        // Your decryption implementation
+        todo!()
+    }
+}
+```
+
+### Encrypting Data
+
+Once you have a [`Cipher`] implementation, you can use the [`Encrypt`] trait to encrypt various types:
+
+```rust
+use vitaminc_aead::{Encrypt, Cipher};
+
+fn encrypt_data<C: Cipher>(cipher: &C) -> Result<(), Box<dyn std::error::Error>> {
+    // Encrypt a string
+    let encrypted_string = "secret message".encrypt(cipher)?;
+
+    // Encrypt with additional authenticated data
+    let encrypted_with_aad = "secret".encrypt_with_aad(cipher, "context data")?;
+
+    // Encrypt a byte array
+    let data = [1, 2, 3, 4, 5];
+    let encrypted_bytes = data.encrypt(cipher)?;
+
+    Ok(())
+}
+```
+
+### Decrypting Data
+
+Use the [`Decrypt`] trait to decrypt ciphertext back to the original type:
+
+```rust
+use vitaminc_aead::{Decrypt, Cipher, LocalCipherText};
+
+fn decrypt_data<C: Cipher>(
+    cipher: &C,
+    ciphertext: LocalCipherText
+) -> Result<String, Box<dyn std::error::Error>> {
+    // Decrypt back to a String
+    let plaintext = String::decrypt(ciphertext, cipher)?;
+    Ok(plaintext)
+}
+```
+
+### Additional Authenticated Data (AAD)
+
+Many types can be used as AAD through the [`IntoAad`] trait:
+
+```rust
+# fn main() -> Result<(), Box<dyn std::error::Error>> {
+use vitaminc_aead::{Encrypt, Cipher};
+use vitaminc_encrypt::{Key, Aes256Cipher};
+use vitaminc_protected::Protected;
+use vitaminc_random::{Generatable, SafeRand, SeedableRng};
+
+let key = Key::random(&mut SafeRand::from_entropy())?;
+let cipher = Aes256Cipher::new(&key)?;
+
+// Use a string as AAD
+"my-secret".encrypt_with_aad(&cipher, "user_id:123")?;
+
+// Use a byte slice as AAD
+"my-secret".encrypt_with_aad(&cipher, &b"metadata"[..])?;
+
+// Use a u64 as AAD
+"my-secret".encrypt_with_aad(&cipher, 42u64)?;
+
+// Use a tuple to combine multiple AAD values
+"my-secret".encrypt_with_aad(&cipher, ("user_id", "session_token"))?;
+
+// Use no AAD
+"my-secret".encrypt_with_aad(&cipher, ())?;
+# Ok(())
+# }
```

Comment:
What are these lines starting with `#`? Do they have a special meaning?

Reply:
@yujiyokoo In code examples in Rust doc the `#` means "include the line for compilation & testing the docs but do not render that bit of the example in the rendered version".

Reply:
It mean you can hide boilerplate, but still compile & test real working code.

Reply:
Great. Thanks 🙏